File: output/ChatGPT/avatar/Java/C++/atcoder_ABC136_B.cpp
#include <iostream>
#include <cmath>

int main() {
    int N;
    std::cin >> N;
    int count = 0;
    int temp = N;
    int ans = 0;
    
    while (temp > 0) {
        temp /= 10;
        count++;
    }
    
    for (int i = 1; i <= count; i++) {
        if (i == count) {
            if (i % 2 == 1) {
                ans += (N - std::pow(10, i - 1) + 1);
            }
        } else {
            if (i == 1) {
                ans += 9;
            } else if (i % 2 == 1) {
                ans += (std::pow(10, i) - std::pow(10, i - 1));
            }
        }
    }
    
    std::cout << ans << std::endl;
    
    return 0;
}

File: output/CodeLlama/avatar/Java/Python/atcoder_ABC136_B.py
import math
import sys

N = int(sys.stdin.readline().rstrip())
count = 0
temp = N
ans = 0
while temp > 0:
    temp //= 10
    count += 1
for i in range(1, count + 1):
    if i == count:
        if i % 2 == 1:
            ans += (N - math.pow(10, i - 1) + 1)
    else:
        if i == 1:
            ans += 9
        elif i % 2 == 1:
            ans += (math.pow(10, i) - math.pow(10, i - 1))
print(ans)